packages/svelte/src/internal/client/dom/elements/bindings/select.js

```diff
@@ -98,15 +99,23 @@ export function bind_select_value(select, get_value, update) {
 			// @ts-ignore
 			var value = select.__value;
 			select_option(select, value, mounting);
-			/** @type {HTMLOptionElement | null} */
-			var selected_option = select.querySelector(':checked');
-			if (selected_option === null || get_option_value(selected_option) !== value) {
-				update('');
+			if (select.multiple) {
+				var selected_options = [].map.call(select.querySelectorAll(':checked'), get_option_value);
+				if (is_array(value) && selected_options.every((e) => value.includes)) {
```

Comment:
This is always truthy: `(e) => value.includes`
Also, wasn't the check meant to be negated and needs to be the other way around:
There is some value that is not found in the current set of selected options.
```suggestion
				if (is_array(value) &&
					value.some((e) => selected_options.includes(e) == false)) {
```

Also, for large numbers of options this `O(n^2)` sort of logic is probably not great.
- Is the check even necessary; would always updating have adverse effects?
- Maybe construct a `Set` and use `has` to reduce complexity to `O(n)`?